core/server/services/slack.js

```diff
@@ -48,11 +54,51 @@ function ping(post) {
             return;
         }
 
+        var blogUrl = config.get('url');
+        var blogTitle = settingsCache.get('title');
+
         slackData = {
-            text: message,
+            // We are handling the case of test notification here by checking
+            // if it is a post or a test message to check webhook working.
+            text: schema.isPost(post) ? `Notification from *${blogTitle}* :ghost:` : message,
             unfurl_links: true,
             icon_url: imageLib.blogIcon.getIconUrl(true),
-            username: 'Ghost'
+            username: 'Ghost',
+            // We don't want to send attachment if it is a test notification.
+            attachments: schema.isPost(post) ? [
+                {
+                    fallback: 'Sorry, content cannot be shown.',
+                    title: title,
+                    title_link: message,
+                    author_name: blogTitle,
+                    image_url: post && post.feature_image ? `${blogUrl}${post.feature_image}` : null,
+                    color: '#008952',
+                    fields: [
+                        {
+                            title: 'Description',
+                            value: post.html ? `${post.html.replace(/<[^>]+>/g, '').split('.').slice(0, 3).join('.')}.` : '',
```

Comment:
This seems to be doing what `excerpt` helper is for: https://github.com/TryGhost/Ghost/blob/master/core/server/helpers/excerpt.js#L15 . Think it would be better keeping things consistent: using  `custom_excerpt` if it's present in the post, or falling back to html content otherwise with [cleanup](https://github.com/TryGhost/Ghost/blob/master/core/server/data/meta/excerpt.js#L3) in the end